### PR TITLE
Inspector fixup

### LINKED ISSUE #45: Inspectors
A list of bugs(?) and missing things for inspectors before we can use revm in Foundry:

- It seems that `call_end` is called instead of `create_end`, which is never called
- We should add more parameters to `call_end` and `create_end` (such as gas left after the call, return values etc.)
- It seems like `Inspector::call` is only called for the upper-most call(? not sure on this), not for internal calls

### PR DESCRIPTION
**Pull request by onbjerg**
An attempt at fixing #45 

- Adds parameters to `create_end` and `call_end`
- Calls `create_end` and `call_end` in the correct places

Putting this in draft to get comments - is this the right direction and/or does this break anything?

Closes #45

### CODE REVIEW

File: crates/revm/src/evm_impl.rs [hunk 1]
```diff
@@ -586,42 +583,84 @@ impl<'a, GSPEC: Spec, DB: Database + 'a, const INSPECT: bool> Host
...
+                &init_code,
+                gas_limit,
+            );
             if ret != Return::Continue {
-                return (ret, gas, bytes, out);
+                return (ret, address, gas, out);
             }
         }
-        self.create_inner::<SPEC>(caller, scheme, value, init_code, gas)
+        let (ret, address, gas, out) =
```
Comment (onbjerg) [inline crates/revm/src/evm_impl.rs:601]: A note here, it seems that `out` is always `Bytes::empty`?

**Comment by rakita** [comment]
Nice, this is mostly what is missng. I would like to check few more things over the weekend, but at first look it looks great.

btw you dont need to create issue every time, just creating pull request is enought,

File: crates/revm/src/evm_impl.rs (hunk 1 above)
Comment (rakita) [reply crates/revm/src/evm_impl.rs:601]: It most cases it is. Only here it returns something: https://github.com/bluealloy/revm/blob/c0bab703000e125a81b2ef35210af55a56d5664f/crates/revm/src/evm_impl.rs#L398

**Review by rakita — APPROVED** [review]
will merge after ci finishes

**Comment by rakita** [comment]
I will fix later this monster number of inputs for create/call functions

**Pull request merged**